api/overdrive.py

```diff
@@ -819,23 +819,22 @@ def circulation_lookup(self, book):
 
     def update_formats(self, licensepool):
         """Update the format information for a single book.
+
+        Incidentally updates the metadata, just in case Overdrive has
+        changed it.
         """
         info = self.metadata_lookup(licensepool.identifier)
 
         metadata = OverdriveRepresentationExtractor.book_info_to_metadata(
-            info, include_bibliographic=False, include_formats=True)
+            info, include_bibliographic=True, include_formats=True)
         if not metadata:
             # No work to be done.
             return
-        circulation_data = metadata.circulation
 
-        # The identifier in the CirculationData needs to match the
-        # identifier associated with the LicensePool -- otherwise
-        # a new LicensePool will be created.
-        circulation_data._primary_identifier.identifier = licensepool.identifier.identifier
-        replace = ReplacementPolicy(formats=True)
-        _db = Session.object_session(licensepool)
-        circulation_data.apply(_db, licensepool.collection, replace)
+        edition, ignore = self._edition(licensepool)
+
+        replace = ReplacementPolicy.from_license_source(self._db)
```

Comment:
What is this replacing?

Reply:
The old code is `replace = ReplacementPolicy(formats=True)`, which will replace an old list of delivery mechanisms with a new list. This replaces basically everything, since a license source is considered authoritative for everything.